diff --git a/stdlib/public/core/FloatingPointTypes.swift.gyb b/stdlib/public/core/FloatingPointTypes.swift.gyb
index db311b34abd95995b5512d8175c3f83055146ce0..cbd72c2d2680622adb11997ec59ea2d9242ef13f 100644
--- a/stdlib/public/core/FloatingPointTypes.swift.gyb
+++ b/stdlib/public/core/FloatingPointTypes.swift.gyb
@@ -54,7 +54,7 @@ else:
 }%
 
 % if bits == 80:
-#if !os(Windows) && (arch(i386) || arch(x86_64))
+#if !(os(Windows) || os(Android)) && (arch(i386) || arch(x86_64))
 % end
 
 ${SelfDocComment}
@@ -1655,7 +1655,7 @@ extension ${Self} {
 %   That = src_type.stdlib_name
 
 %   if (srcBits == 80) and (bits != 80):
-#if !os(Windows) && (arch(i386) || arch(x86_64))
+#if !(os(Windows) || os(Android)) && (arch(i386) || arch(x86_64))
 %   end
 
 %   if srcBits == bits:
