diff --git a/stdlib/public/core/MathFunctions.swift.gyb b/stdlib/public/core/MathFunctions.swift.gyb
index 790539c69507de8267d5213fc5194cc45e24dc88..7bb3bac2d9efb9f4de99caeb1bcd1da21cd952b3 100644
--- a/stdlib/public/core/MathFunctions.swift.gyb
+++ b/stdlib/public/core/MathFunctions.swift.gyb
@@ -62,7 +62,7 @@ public protocol ElementaryFunctions {
 
 %for type in all_floating_point_types():
 % if type.bits == 80:
-#if (arch(i386) || arch(x86_64)) && !os(Windows)
+#if (arch(i386) || arch(x86_64)) && !(os(Windows) || os(Android))
 % end
 % Self = type.stdlib_name
 extension ${Self}: ElementaryFunctions {
