diff --git a/include/swift/Runtime/SwiftDtoa.h b/include/swift/Runtime/SwiftDtoa.h
index c309d2e97985ef70d6c714ca802e00831f113fe9..07670af400c2b6a137b18b73b34406dd41002a8e 100644
--- a/include/swift/Runtime/SwiftDtoa.h
+++ b/include/swift/Runtime/SwiftDtoa.h
@@ -37,6 +37,10 @@
  #endif
 #endif
 
+#if defined(__ANDROID__) && (defined(__i386) || defined(__x86_64__))
+ #undef SWIFT_DTOA_FLOAT80_SUPPORT
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
