diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd358d45ae41b3453ebf98a52d95a88a77099436..c87a46a842176ee3d0a4b1531528ca1d614da3c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -287,6 +287,8 @@ add_swift_library(Foundation
                     $<$<PLATFORM_ID:Windows>:-liphlpapi>
                     $<$<PLATFORM_ID:Windows>:-lpathcch>
                     $<$<PLATFORM_ID:Windows>:$<TARGET_OBJECTS:CoreFoundationResources>>
+                    ${ADDITIONAL_SWIFT_LINK_FLAGS}
+                    -Xlinker --defsym -Xlinker '__CFConstantStringClassReference=$$s10Foundation19_NSCFConstantStringCN'
                   SWIFT_FLAGS
                     -DDEPLOYMENT_RUNTIME_SWIFT
                     $<$<BOOL:FOUNDATION_ENABLE_LIBDISPATCH>:-DDEPLOYMENT_ENABLE_LIBDISPATCH>
@@ -295,6 +297,7 @@ add_swift_library(Foundation
                     $<$<BOOL:ENABLE_TESTING>:-enable-testing>
                     $<$<NOT:$<CONFIG:Debug>>:-O>
                     ${swift_libc_flags}
+                    ${ADDITIONAL_SWIFT_FLAGS}
                   DEPENDS
                     uuid
                     CoreFoundation
@@ -353,6 +356,7 @@ add_swift_library(FoundationNetworking
                     ${Foundation_RPATH}
                     ${WORKAROUND_SR9138}
                     ${WORKAROUND_SR9995}
+                    ${ADDITIONAL_SWIFT_LINK_FLAGS}
                   SWIFT_FLAGS
                     -DDEPLOYMENT_RUNTIME_SWIFT
                     -DNS_BUILDING_FOUNDATION_NETWORKING
@@ -361,6 +365,7 @@ add_swift_library(FoundationNetworking
                     ${libdispatch_cflags}
                     ${swift_enable_testing}
                     ${swift_optimization_flags}
+                    ${ADDITIONAL_SWIFT_FLAGS}
                   DEPENDS
                     uuid
                     CoreFoundation
@@ -381,6 +386,7 @@ if(NOT BUILD_SHARED_LIBS)
       ${LIBXML2_LIBRARIES})
 endif()
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
 add_swift_executable(plutil
                      SOURCES
                        Tools/plutil/main.swift
@@ -406,6 +412,7 @@ add_swift_executable(plutil
                        uuid
                        Foundation
                        CoreFoundation)
+endif()
 
 if(ENABLE_TESTING)
   add_swift_executable(xdgTestHelper
@@ -665,7 +672,9 @@ install(FILES
           CoreFoundation/URL.subproj/module.map
         DESTINATION
           lib/swift/CFURLSessionInterface)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
 install(PROGRAMS
           ${CMAKE_CURRENT_BINARY_DIR}/plutil${CMAKE_EXECUTABLE_SUFFIX}
         DESTINATION
           ${CMAKE_INSTALL_FULL_BINDIR})
+endif()
