diff --git a/CMakeLists.txt b/CMakeLists.txt
index c5dbd1e67146090a8ce4a9f49d1c8a53f9c63a1d..f43d69fc9e542ee307f3bca7ebe0bb348c6a63cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -292,6 +292,8 @@ add_swift_library(Foundation
                     $<$<PLATFORM_ID:Windows>:-liphlpapi>
                     $<$<PLATFORM_ID:Windows>:-lpathcch>
                     $<$<PLATFORM_ID:Windows>:$<TARGET_OBJECTS:CoreFoundationResources>>
+                    ${ADDITIONAL_SWIFT_LINK_FLAGS}
+                    -Xlinker --defsym -Xlinker '__CFConstantStringClassReference=$$s10Foundation19_NSCFConstantStringCN'
                   SWIFT_FLAGS
                     -DDEPLOYMENT_RUNTIME_SWIFT
                     $<$<BOOL:FOUNDATION_ENABLE_LIBDISPATCH>:-DDEPLOYMENT_ENABLE_LIBDISPATCH>
@@ -300,6 +302,7 @@ add_swift_library(Foundation
                     ${swift_enable_testing}
                     ${swift_optimization_flags}
                     ${swift_libc_flags}
+                    ${ADDITIONAL_SWIFT_FLAGS}
                   DEPENDS
                     uuid
                     CoreFoundation
@@ -358,6 +361,7 @@ add_swift_library(FoundationNetworking
                     ${Foundation_RPATH}
                     ${WORKAROUND_SR9138}
                     ${WORKAROUND_SR9995}
+                    ${ADDITIONAL_SWIFT_LINK_FLAGS}
                   SWIFT_FLAGS
                     -DDEPLOYMENT_RUNTIME_SWIFT
                     -DNS_BUILDING_FOUNDATION_NETWORKING
@@ -366,6 +370,7 @@ add_swift_library(FoundationNetworking
                     ${libdispatch_cflags}
                     ${swift_enable_testing}
                     ${swift_optimization_flags}
+                    ${ADDITIONAL_SWIFT_FLAGS}
                   DEPENDS
                     uuid
                     CoreFoundation
@@ -386,6 +391,7 @@ if(NOT BUILD_SHARED_LIBS)
       ${LIBXML2_LIBRARIES})
 endif()
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
 add_swift_executable(plutil
                      SOURCES
                        Tools/plutil/main.swift
@@ -409,6 +415,7 @@ add_swift_executable(plutil
                        uuid
                        Foundation
                        CoreFoundation)
+endif()
 
 if(ENABLE_TESTING)
   add_swift_executable(xdgTestHelper
@@ -644,7 +651,10 @@ install(FILES
           CoreFoundation/Base.subproj/module.map
         DESTINATION
           lib/swift/CoreFoundation)
+
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
 install(PROGRAMS
           ${CMAKE_CURRENT_BINARY_DIR}/plutil${CMAKE_EXECUTABLE_SUFFIX}
         DESTINATION
           ${CMAKE_INSTALL_FULL_BINDIR})
+endif()
