diff --git a/CMakeLists.txt b/CMakeLists.txt
index ecc6c3c9e30c51e70c1c7ed2f2bd074befd824e3..f910d88d2eace7750a0428c9850ef86fe6496a96 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,14 @@ set(BUILD_SHARED_LIBS NO)
 add_subdirectory(CoreFoundation EXCLUDE_FROM_ALL)
 set(BUILD_SHARED_LIBS ${SAVED_BUILD_SHARED_LIBS})
 
+# User-configurable Android specific options.
+set(SWIFT_ANDROID_API_LEVEL "" CACHE STRING
+  "Version number for the Android API")
+set(SWIFT_ANDROID_NDK_PATH "" CACHE STRING
+  "Path to the directory that contains the Android NDK tools that are executable on the build machine")
+set(SWIFT_ANDROID_NDK_GCC_VERSION "" CACHE STRING
+  "The GCC version to use when building for Android. Currently only 4.9 is supported.")
+
 # Setup include paths for uuid/uuid.h
 target_include_directories(CoreFoundation PRIVATE
   ${CMAKE_CURRENT_SOURCE_DIR}
@@ -50,7 +58,9 @@ target_include_directories(CoreFoundation PRIVATE
 
 add_subdirectory(uuid)
 add_subdirectory(Foundation)
-add_subdirectory(Tools)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
+  add_subdirectory(Tools)
+endif()
 if(ENABLE_TESTING)
   add_subdirectory(TestFoundation)
 endif()
