diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff47064da305032b285037bbcd6846bed15c027c..28f63dabaee5234e177ff113beb425b413e8bdfe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,7 @@ option(FOUNDATION_PATH_TO_XCTEST_BUILD "Path to XCTest build" "")
 find_package(CURL REQUIRED)
 find_package(ICU COMPONENTS uc i18n REQUIRED)
 find_package(LibXml2 REQUIRED)
-find_package(UUID REQUIRED)
+# find_package(UUID REQUIRED)
 
 include(SwiftSupport)
 include(GNUInstallDirs)
@@ -45,6 +45,7 @@ ExternalProject_Add(CoreFoundation
                       -DCF_PATH_TO_LIBDISPATCH_SOURCE=${FOUNDATION_PATH_TO_LIBDISPATCH_SOURCE}
                       -DCF_PATH_TO_LIBDISPATCH_BUILD=${FOUNDATION_PATH_TO_LIBDISPATCH_BUILD}
                       -DICU_INCLUDE_DIR=${ICU_INCLUDE_DIR}
+                      -DICU_LIBRARY=${ICU_LIBRARY}
                       -DLIBXML2_LIBRARY=${LIBXML2_LIBRARY}
                       -DLIBXML2_INCLUDE_DIR=${LIBXML2_INCLUDE_DIR}
                       -DCURL_LIBRARY=${CURL_LIBRARY}
