diff --git a/CMakeLists.txt b/CMakeLists.txt
index efe6aac25c59c9de17a56b58d7a7ab4fa39e1047..a3adaf02a3761bf58b9463d0dd256968743621aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,10 @@ ExternalProject_Add(CoreFoundation
                     CMAKE_COMMAND
                       ${CMAKE_COMMAND}
                     CMAKE_ARGS
+                      -DANDROID_PLATFORM=${ANDROID_PLATFORM}
+                      -DANDROID_ABI=${ANDROID_ABI}
+                      -DANDROID_TOOLCHAIN=${ANDROID_TOOLCHAIN}
+                      -DANDROID_STL=${ANDROID_STL}
                       -DBUILD_SHARED_LIBS=NO
                       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                       -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
@@ -308,6 +312,7 @@ add_swift_library(Foundation
                     ${deployment_enable_libdispatch}
                     -F${install_dir}/System/Library/Frameworks
                     -D_DLL
+                    ${ADDITIONAL_SWIFT_CFLAGS}
                   LINK_FLAGS
                     -L${install_dir}/usr/lib
                     -lCoreFoundation
@@ -318,6 +323,7 @@ add_swift_library(Foundation
                     -L${CMAKE_CURRENT_BINARY_DIR}
                     -luuid
                     ${Foundation_rpath_flags}
+                    ${ADDITIONAL_SWIFT_LINK_FLAGS}
                   SWIFT_FLAGS
                     -DDEPLOYMENT_RUNTIME_SWIFT
                     ${deployment_enable_libdispatch}
@@ -325,6 +331,7 @@ add_swift_library(Foundation
                     ${libdispatch_cflags}
                     ${swift_enable_testing}
                     ${swift_optimization_flags}
+                    ${ADDITIONAL_SWIFT_FLAGS}
                   DEPENDS
                     CoreFoundation)
 
@@ -340,6 +347,7 @@ if(NOT BUILD_SHARED_LIBS)
       ${LIBXML2_LIBRARIES})
 endif()
 
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
 add_swift_executable(plutil
                      SOURCES
                        Tools/plutil/main.swift
@@ -365,6 +373,7 @@ add_swift_executable(plutil
                        uuid
                        Foundation
                        CoreFoundation)
+endif()
 
 if(ENABLE_TESTING)
   add_swift_executable(xdgTestHelper
@@ -575,7 +584,9 @@ install(FILES
           CoreFoundation/Base.subproj/module.map
         DESTINATION
           lib/swift/CoreFoundation)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
 install(PROGRAMS
           ${CMAKE_CURRENT_BINARY_DIR}/plutil
         DESTINATION
           ${CMAKE_INSTALL_FULL_BINDIR})
+endif()
