diff --git a/CoreFoundation/CMakeLists.txt b/CoreFoundation/CMakeLists.txt
index 0aa9565e3c3930e7375cc548870003c1b8432653..f697862f4d970e628cf9aa896cb38f8f9357411a 100644
--- a/CoreFoundation/CMakeLists.txt
+++ b/CoreFoundation/CMakeLists.txt
@@ -405,9 +405,9 @@ if(CF_DEPLOYMENT_SWIFT)
                            PRIVATE
                              /clang:-fcf-runtime-abi=swift)
   else()
-    target_compile_options(CoreFoundation
-                           PRIVATE
-                             -fcf-runtime-abi=swift)
+    # target_compile_options(CoreFoundation
+    #                        PRIVATE
+    #                          -fcf-runtime-abi=swift)
   endif()
 endif()
 
