diff --git a/Foundation/FileManager.swift b/Foundation/FileManager.swift
index c51536246fc92efd82115c2217b256466b234712..a537ae8e676f4e5326d8cd4aae38a2e312b42faf 100644
--- a/Foundation/FileManager.swift
+++ b/Foundation/FileManager.swift
@@ -91,7 +91,10 @@ open class FileManager : NSObject {
             urls = mountPoints(statBuf, Int(fsCount))
         }
 #else
+/* FIXME: Android compile Workaround.
 #error("Requires a platform-specific implementation")
+*/
+        urls = [] // FIXME: Android compile Workaround.
 #endif
         return urls
     }
@@ -581,6 +584,7 @@ open class FileManager : NSObject {
         var entry = dirent()
         var result: UnsafeMutablePointer<dirent>? = nil
 
+        /* FIXME: Android compile Workaround.
         while readdir_r(dir, &entry, &result) == 0 {
             guard result != nil else {
                 return
@@ -595,6 +599,7 @@ open class FileManager : NSObject {
                 try closure(entryName, entryType)
             }
         }
+        */
     }
 
     /**
@@ -651,6 +656,8 @@ open class FileManager : NSObject {
         This method replaces fileAttributesAtPath:traverseLink:.
      */
     open func attributesOfItem(atPath path: String) throws -> [FileAttributeKey : Any] {
+        return [:]
+        /* FIXME: Android compile Workaround.
         let s = try _lstatFile(atPath: path)
         var result = [FileAttributeKey : Any]()
         result[.size] = NSNumber(value: UInt64(s.st_size))
@@ -698,6 +705,7 @@ open class FileManager : NSObject {
         result[.groupOwnerAccountID] = NSNumber(value: UInt64(s.st_gid))
         
         return result
+        */
     }
     
     /* attributesOfFileSystemForPath:error: returns an NSDictionary of key/value pairs containing the attributes of the filesystem containing the provided path. If this method returns 'nil', an NSError will be returned by reference in the 'error' parameter. This method does not traverse a terminal symlink.
@@ -880,6 +888,7 @@ open class FileManager : NSObject {
                 return
         }
 
+        /* FIXME: Android compile Workaround.
         let fileType = FileAttributeType(statMode: stat.st_mode)
         if fileType == .typeDirectory {
             try createDirectory(atPath: dstPath, withIntermediateDirectories: false, attributes: nil)
@@ -903,6 +912,7 @@ open class FileManager : NSObject {
         } else {
             try body(srcPath, dstPath, fileType)
         }
+        */
     }
     
     private func shouldProceedAfterError(_ error: Error, copyingItemAtPath path: String, toPath: String, isURL: Bool) -> Bool {
@@ -1413,8 +1423,10 @@ open class FileManager : NSObject {
 
         if file1Type == S_IFCHR || file1Type == S_IFBLK {
             // For character devices, just check the major/minor pair is the same.
+            /* FIXME: Android compile Workaround.
             return _dev_major(file1.st_rdev) == _dev_major(file2.st_rdev)
                 && _dev_minor(file1.st_rdev) == _dev_minor(file2.st_rdev)
+            */
         }
 
         // If both paths point to the same device/inode or they are both zero length
