diff --git a/uuid/uuid.h b/uuid/uuid.h
index cc997a855b5484ff33a2cfc87698308d6829dca5..a23eb95eaf383738da5d225c2cee0de7fb325e38 100644
--- a/uuid/uuid.h
+++ b/uuid/uuid.h
@@ -33,7 +33,7 @@
 #ifndef _UUID_UUID_H
 #define _UUID_UUID_H
 
-#include <TargetConditionals.h>
+#include "../CoreFoundation/Base.subproj/SwiftRuntime/TargetConditionals.h"
 #if TARGET_OS_MAC
 #include <sys/_types.h>
 #include <sys/_types/_uuid_t.h>
