diff --git a/CoreFoundation/CMakeLists.txt b/CoreFoundation/CMakeLists.txt
index 0928546c44ead594701ce3a8059e54b5d96b0a0f..cd5160fda695895e57c03184c7fb24b3e1ac432b 100644
--- a/CoreFoundation/CMakeLists.txt
+++ b/CoreFoundation/CMakeLists.txt
@@ -348,12 +348,21 @@ if(CMAKE_SYSTEM_NAME STREQUAL Windows)
   add_dependencies(CoreFoundation CoreFoundationWindowsTimeZonesPLists)
 endif()
 if(CMAKE_SYSTEM_NAME STREQUAL Linux OR CMAKE_SYSTEM_NAME STREQUAL Android)
+  if(CMAKE_SYSTEM_NAME STREQUAL Android)
   target_compile_definitions(CoreFoundation
                              PRIVATE
-                               -D_GNU_SOURCE)
+                               -D__ANDROID__)
   target_compile_definitions(CFURLSessionInterface
                              PRIVATE
-                               -D_GNU_SOURCE)
+                               -D__ANDROID__)
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Linux)
+     target_compile_definitions(CoreFoundation
+                                PRIVATE
+                                  -D_GNU_SOURCE)
+     target_compile_definitions(CFURLSessionInterface
+                                PRIVATE
+                                  -D_GNU_SOURCE)
+  endif()
   if(HAVE_SCHED_GETAFFINITY)
     target_compile_definitions(CoreFoundation
                                PRIVATE
@@ -438,14 +447,17 @@ target_include_directories(CFURLSessionInterface
                            PRIVATE
                              ${CF_PATH_TO_LIBDISPATCH_SOURCE}
                              ${CF_PATH_TO_LIBDISPATCH_BUILD}/tests)
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
+
+target_include_directories(CoreFoundation PRIVATE ${CMAKE_SOURCE_DIR})
+target_include_directories(CoreFoundation PRIVATE ${CMAKE_SOURCE_DIR}/CoreFoundation/Base.subproj/SwiftRuntime)
+# if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
   target_include_directories(CoreFoundation
                              SYSTEM PRIVATE
                                ${CF_PATH_TO_LIBDISPATCH_SOURCE}/src/BlocksRuntime)
   target_include_directories(CFURLSessionInterface
                              SYSTEM PRIVATE
                                ${CF_PATH_TO_LIBDISPATCH_SOURCE}/src/BlocksRuntime)
-endif()
+# endif()
 
 if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")
   target_compile_options(CoreFoundation
@@ -497,12 +509,12 @@ if(CF_DEPLOYMENT_SWIFT)
                            PRIVATE
                              /clang:-fcf-runtime-abi=swift)
   else()
-    target_compile_options(CoreFoundation
-                           PRIVATE
-                             -fcf-runtime-abi=swift)
-    target_compile_options(CFURLSessionInterface
-                           PRIVATE
-                             -fcf-runtime-abi=swift)
+    # target_compile_options(CoreFoundation
+    #                        PRIVATE
+    #                          -fcf-runtime-abi=swift)
+    # target_compile_options(CFURLSessionInterface
+    #                        PRIVATE
+    #                          -fcf-runtime-abi=swift)
   endif()
 endif()
 
@@ -589,6 +601,9 @@ install(TARGETS
 		  CFURLSessionInterface
         DESTINATION
           "${CMAKE_INSTALL_FULL_LIBDIR}")
+
+string(REGEX REPLACE "/$" "" CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
+
 install(DIRECTORY
           ${CoreFoundation_FRAMEWORK_DIRECTORY}
         DESTINATION
