diff --git a/stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb b/stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb
index 1b8823fa9a8bed79a76970634b454d273d94ecf1..1e6a99c1d78bdf0f6fd0fda59dd7d7def16a9313 100644
--- a/stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb
+++ b/stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb
@@ -46,7 +46,7 @@ public struct CGFloat {
     self.native = NativeType(value)
   }
 
-#if !os(Windows) && (arch(i386) || arch(x86_64))
+#if !(os(Windows) || os(Android)) && (arch(i386) || arch(x86_64))
   @_transparent public init(_ value: Float80) {
     self.native = NativeType(value)
   }
