diff --git a/CoreFoundation/Base.subproj/CFKnownLocations.c b/CoreFoundation/Base.subproj/CFKnownLocations.c
index 1433c356194776ea7cfe8bacc4f43e143d966951..c4ceec676ab76ddf1aca21e972c0142a00f46396 100644
--- a/CoreFoundation/Base.subproj/CFKnownLocations.c
+++ b/CoreFoundation/Base.subproj/CFKnownLocations.c
@@ -45,7 +45,7 @@ CFURLRef _Nullable _CFKnownLocationCreatePreferencesURLForUser(CFKnownLocationUs
         }
             
     }
-#elif !DEPLOYMENT_RUNTIME_OBJC && !TARGET_OS_WIN32 && !TARGET_OS_ANDROID
+#elif !DEPLOYMENT_RUNTIME_OBJC && !TARGET_OS_WIN32
     
 /*
  Building for an OS that uses the FHS, BSD's hier(7), and/or the XDG specification for paths:
