diff --git a/CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h b/CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h
index 204e04e42a6135489b7fbdf60941d765e2bd70b2..b56fb3b86467a2c788205660f098a3fb3f921c67 100644
--- a/CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h
+++ b/CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h
@@ -55,7 +55,7 @@
 #include <malloc/malloc.h>
 #endif
 
-#if TARGET_OS_LINUX
+#if TARGET_OS_LINUX && !TARGET_OS_ANDROID
 #include <features.h>
 
 #if __GLIBC_PREREQ(2, 28) == 0
@@ -68,6 +68,15 @@
 #endif //__GLIBC_PREREQ(2. 28)
 #endif // TARGET_OS_LINUX
 
+#if TARGET_OS_ANDROID
+#include <features.h>
+// required for statx() system call, glibc >=2.28 wraps the kernel function
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <linux/stat.h>
+#define AT_STATX_SYNC_AS_STAT   0x0000  /* - Do whatever stat() does */
+#endif
 
 _CF_EXPORT_SCOPE_BEGIN
 
