diff --git a/stdlib/public/stubs/CMakeLists.txt b/stdlib/public/stubs/CMakeLists.txt
index 2196d456e791feacf2d5ec6f92d615f5a0ee2bbf..fa00c05f415b55e8e2e30ac6423e6016861ca18f 100644
--- a/stdlib/public/stubs/CMakeLists.txt
+++ b/stdlib/public/stubs/CMakeLists.txt
@@ -37,7 +37,7 @@ add_swift_target_library(swiftStdlibStubs
                   INSTALL_IN_COMPONENT
                     stdlib)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT SWIFT_OVERLAY_TARGETS STREQUAL "")
   set_property(SOURCE
                  SwiftNativeNSXXXBaseARC.m
                APPEND_STRING PROPERTY COMPILE_FLAGS
