diff --git a/stdlib/CMakeLists.txt b/stdlib/CMakeLists.txt
index d05618c94c5e79f9bde808efb7d6faa2082b7fa8..5fb9e7b618c27f12702345c9b9332cf7b102dd8c 100644
--- a/stdlib/CMakeLists.txt
+++ b/stdlib/CMakeLists.txt
@@ -111,7 +111,9 @@ if(SWIFT_STDLIB_ENABLE_SIB_TARGETS)
   swift_create_stdlib_targets("swift-stdlib" "sibopt" TRUE)
   swift_create_stdlib_targets("swift-stdlib" "sibgen" TRUE)
 endif()
-swift_create_stdlib_targets("swift-test-stdlib" "" FALSE)
+if (SWIFT_INCLUDE_TESTS)
+   swift_create_stdlib_targets("swift-test-stdlib" "" FALSE)
+endif()
 
 foreach(sdk ${SWIFT_SDKS})
   foreach(arch ${SWIFT_SDK_${sdk}_ARCHITECTURES})
