diff --git a/stdlib/private/CMakeLists.txt b/stdlib/private/CMakeLists.txt
index fa67fc55b5..dffc340c47 100644
--- a/stdlib/private/CMakeLists.txt
+++ b/stdlib/private/CMakeLists.txt
@@ -2,7 +2,7 @@ if(SWIFT_BUILD_STDLIB OR SWIFT_BUILD_SDK_OVERLAY)
   add_subdirectory(SwiftPrivate)
 endif()
 
-if(SWIFT_BUILD_SDK_OVERLAY)
+if(SWIFT_BUILD_SDK_OVERLAY AND NOT SWIFT_OVERLAY_TARGETS STREQUAL "")
   # SwiftPrivatePthreadExtras makes use of Darwin/Glibc, which is part of the
   # SDK overlay. It can't be built separately from the SDK overlay.
   add_subdirectory(RuntimeUnittest)
