diff --git a/Foundation/CGFloat.swift b/Foundation/CGFloat.swift
index 1bb768df2982ea692bee7a6c0cc62f1c7327f293..5f33c061dcfc6f6b3e4972af3dce20311b4b3488 100644
--- a/Foundation/CGFloat.swift
+++ b/Foundation/CGFloat.swift
@@ -31,7 +31,7 @@ public struct CGFloat {
         self.native = NativeType(value)
     }
     
-#if !os(Windows) && (arch(i386) || arch(x86_64))
+#if !(os(Windows) || os(Android)) && (arch(i386) || arch(x86_64))
     @_transparent public init(_ value: Float80) {
         self.native = NativeType(value)
     }
