diff --git a/Foundation/FileManager.swift b/Foundation/FileManager.swift
index 14636a90f96e83d2b34eaf462591d50995e3ae3a..4137427909fdfc5526339ac12352ced4e9d3394b 100644
--- a/Foundation/FileManager.swift
+++ b/Foundation/FileManager.swift
@@ -7,12 +7,6 @@
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-#if os(Android) // struct stat.st_mode is UInt32
-internal func &(left: UInt32, right: mode_t) -> mode_t {
-    return mode_t(left) & right
-}
-#endif
-
 #if !canImport(Darwin) && !os(FreeBSD)
 // The values do not matter as long as they are nonzero.
 fileprivate let UF_IMMUTABLE: Int32 = 1
@@ -470,7 +464,12 @@ open class FileManager : NSObject {
             result[.groupOwnerAccountName] = name
         }
 
+#if os(Android) && (arch(i386) || arch(arm))
+        let type = FileAttributeType(statMode: UInt16(s.st_mode))
+#else
         let type = FileAttributeType(statMode: s.st_mode)
+#endif
+
 #endif
         result[.type] = type
 
