diff --git a/CoreFoundation/Base.subproj/SwiftRuntime/CoreFoundation.h b/CoreFoundation/Base.subproj/SwiftRuntime/CoreFoundation.h
index a3ebe169bb3eacd598ab804d921c2dc9389d3f60..4ceba2c11b504c20a8dc8144f46824066ae4e265 100644
--- a/CoreFoundation/Base.subproj/SwiftRuntime/CoreFoundation.h
+++ b/CoreFoundation/Base.subproj/SwiftRuntime/CoreFoundation.h
@@ -32,12 +32,12 @@
 #include <locale.h>
 #include <math.h>
 #include <setjmp.h>
+#include <time.h>
 #include <signal.h>
 #include <stddef.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <time.h>
 
 #if __has_include(<netdb.h>)
 #include <netdb.h> // for Host.swift
