diff --git a/Sources/swift-test/CMakeLists.txt b/Sources/swift-test/CMakeLists.txt
index fc4912ef34c0783b93ff58e9c15598dbe30b500f..ef951364b9010687f1dca5f470f717cdbab665b3 100644
--- a/Sources/swift-test/CMakeLists.txt
+++ b/Sources/swift-test/CMakeLists.txt
@@ -11,7 +11,7 @@ add_executable(swift-test
 target_link_libraries(swift-test PRIVATE
   Commands)
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+#if(CMAKE_SYSTEM_NAME STREQUAL Windows)
 install(TARGETS swift-test
   RUNTIME DESTINATION bin)
-endif()
+#endif()
